src/main/java/se/liu/ida/hefquin/cli/modules/ModFederation.java

```diff
@@ -89,147 +88,151 @@ public FederationCatalog getFederationCatalog() {
+				addSPARQLEndpoint(addrStr, vocabMap);
 			}
-			else {
-				throw new IllegalArgumentException();
+			else if ( fd.getRequiredProperty(iface, RDF.type).getObject().equals(FD.TPFInterface) ){
+				final RDFNode addr = fd.getRequiredProperty(iface, FD.exampleFragmentAddress).getObject();
+
+				final String addrStr;
+				if ( addr.isLiteral() ) {
+					addrStr = addr.asLiteral().getLexicalForm();
+				}
+				else if ( addr.isURIResource() ) {
+					addrStr = addr.asResource().getURI();
+				}
+				else {
+					throw new IllegalArgumentException();
+				}
+
+				addTPFServer(addrStr, vocabMap);
 			}
-
-			addNeo4jServer(addrStr);
-		}
-
-		final ResIterator itGraphQL = fd.listResourcesWithProperty(RDF.type, FD.GraphQLEndpointInterface);
-		while ( itGraphQL.hasNext() ) {
-			final Resource iface = itGraphQL.next();
-			final RDFNode addr = fd.getRequiredProperty(iface, FD.endpointAddress).getObject();
-
-			final String addrStr;
-			if ( addr.isLiteral() ) {
-				addrStr = addr.asLiteral().getLexicalForm();
+			else if ( fd.getRequiredProperty(iface, RDF.type).getObject().equals(FD.brTPFInterface) ){
+				final RDFNode addr = fd.getRequiredProperty(iface, FD.exampleFragmentAddress).getObject();
+
+				final String addrStr;
+				if ( addr.isLiteral() ) {
+					addrStr = addr.asLiteral().getLexicalForm();
+				}
+				else if ( addr.isURIResource() ) {
+					addrStr = addr.asResource().getURI();
+				}
+				else {
+					throw new IllegalArgumentException();
+				}
+
+				addBRTPFServer(addrStr, vocabMap);
 			}
-			else if ( addr.isURIResource() ) {
-				addrStr = addr.asResource().getURI();
+			else if ( fd.getRequiredProperty(iface, RDF.type).getObject().equals(FD.BoltInterface) ){
+				final RDFNode addr = fd.getRequiredProperty(iface, FD.endpointAddress).getObject();
+
+				final String addrStr;
+				if ( addr.isLiteral() ) {
+					addrStr = addr.asLiteral().getLexicalForm();
+				}
+				else if ( addr.isURIResource() ) {
+					addrStr = addr.asResource().getURI();
+				}
+				else {
+					throw new IllegalArgumentException();
+				}
+
+				addNeo4jServer(addrStr, vocabMap);
 			}
-			else {
-				throw new IllegalArgumentException();
+			else if ( fd.getRequiredProperty(iface, RDF.type).getObject().equals(FD.GraphQLEndpointInterface) ){
+				final RDFNode addr = fd.getRequiredProperty(iface, FD.endpointAddress).getObject();
+
+				final String addrStr;
+				if ( addr.isLiteral() ) {
+					addrStr = addr.asLiteral().getLexicalForm();
+				}
+				else if ( addr.isURIResource() ) {
+					addrStr = addr.asResource().getURI();
+				}
+				else {
+					throw new IllegalArgumentException();
+				}
+
+				addGraphQLServer(addrStr);
 			}
 
-			addGraphQLServer(addrStr);
 		}
 	}
 
 	protected void addSPARQLEndpoints( final List<String> sparqlEndpointValues ) {
 		for ( final String v : sparqlEndpointValues )
-			addSPARQLEndpoint(v);
+			// TODO: extend command line to support connect the endpoint to a file that describe the vocabulary mappings
```

Comment:
```suggestion
```

No. We should move away from exposing every configuration detail via a separate CLI argument. Instead, the way to specify the federation to be queried (including corresponding vocab.mappings, if any) should be via the federation description file.